kymatio/scattering2d/backend/tensorflow_backend.py

```diff
@@ -0,0 +1,176 @@
+            FFT^{-1}(res)[u1, u2] = FFT^{-1}(x)[u1 * (2**k), u2 * (2**k)]
+    """
+    def __call__(self, x, k):
+        y = tf.reshape(x,(-1, k, x.shape[1] // k, k, x.shape[2] // k))
+        
+        out = tf.reduce_mean(y, axis=(1, 3))
+        return out
+
+
+class Modulus(object):
+    """
+        This class implements a modulus transform for complex numbers.
+        Usage
+        -----
+        modulus = Modulus()
+        x_mod = modulus(x)
+        Parameters
+        ---------
+        x: input complex tensor.
+        Returns
+        -------
+        output: a real tensor equal to the modulus of x.
+    """
+    def __call__(self, x):
+        norm = tf.abs(x)
+        return tf.cast(norm, tf.complex64)
+
+
+def fft(x, direction='C2C', inverse=False):
+    """
+        Interface with torch FFT routines for 2D signals.
+        Example
+        -------
+        x = torch.randn(128, 32, 32, 2)
+        x_fft = fft(x, inverse=True)
+        Parameters
+        ----------
+        input : tensor
+            complex input for the FFT
+        direction : string
+            'C2R' for complex to real, 'C2C' for complex to complex
+        inverse : bool
+            True for computing the inverse FFT.
+            NB : if direction is equal to 'C2R', then an error is raised.
+    """
+    if direction == 'C2R':
+        if not inverse:
+            raise RuntimeError('C2R mode can only be done with an inverse FFT.')
+
+    if direction == 'C2R':
+        output = tf.math.real(tf.signal.ifft2d(x, name='irfft2d'))*tf.cast(x.shape[-1]*x.shape[-2],tf.float32)
+    elif direction == 'C2C':
+        if inverse:
+            output = tf.signal.ifft2d(x, name='ifft2d')*tf.cast(x.shape[-1]*x.shape[-2], tf.complex64)
+        else:
+            output = tf.signal.fft2d(x, name='fft2d')
+
+    return output
+
+
+def cdgmm(A, B, inplace=False):
+    """
+        Complex pointwise multiplication between (batched) tensor A and tensor B.
+        Parameters
+        ----------
+        A : tensor
+            A is a complex tensor of size (B, C, M, N, 2)
+        B : tensor
+            B is a complex tensor of size (M, N) or real tensor of (M, N)
+        inplace : boolean, optional
+            if set to True, all the operations are performed inplace
+        Returns
+        -------
+        C : tensor
+            output tensor of size (B, C, M, N, 2) such that:
+            C[b, c, m, n, :] = A[b, c, m, n, :] * B[m, n, :]
+    """
+    if B.ndim != 2:
+        raise RuntimeError('The dimension of the second input must be 2.')
+    
+    if not _iscomplex(A):
+        raise TypeError('The first input must be complex.')
+    
+    if not _iscomplex(B) and not _isreal(B):
+        raise TypeError('The second input must be complex or real.')
+
+    if A.shape[-2:] != B.shape[-2:]:
+        raise RuntimeError('The inputs are not compatible for '
+                           'multiplication.')
```

Comment:
I like the wording on this.

Reply:
So... do you want to change it?

Reply:
@MuawizChaudhary you want or you don't want

Reply:
No, this is good. Just a positive comment :)